frontend/app/src/components/accounts/balances/NonFungibleBalances.vue

```diff
@@ -188,18 +190,26 @@ const {
 } = useHistoryPaginationFilter<
   NonFungibleBalance,
   NonFungibleBalancesRequestPayload,
-  NonFungibleBalance
->(null, true, useEmptyFilter, fetchNonFungibleBalances, {
-  onUpdateFilters(query) {
-    set(ignoredAssetsHandling, query.ignoredAssetsHandling || 'exclude');
-  },
-  extraParams,
-  defaultSortBy: {
-    pagination: 'name',
-    pageParams: ['name'],
-    pageParamsAsc: [true]
+  NonFungibleBalance,
+  Collection<NonFungibleBalance>
+>(
+  null,
+  true,
+  useEmptyFilter,
+  fetchNonFungibleBalances,
+  defaultCollectionState,
```

Comment:
Wouldn't it make sense to move the collection state to the options instead and use the default internally only if there is no external state?

Reply:
I did initially, but was causing type errors, but I'll check again.

Reply:
this will require one typecasting... 

```typescript
  const getCollectionDefault = (): S => {
    if (defaultCollection) {
      return defaultCollection();
    }

    return defaultCollectionState<V>() as S;
  };
```

been trying to avoid it, it's pressing my neck. 😛

Reply:
> this will require one typecasting...
> 
> ```ts
>   const getCollectionDefault = (): S => {
>     if (defaultCollection) {
>       return defaultCollection();
>     }
> 
>     return defaultCollectionState<V>() as S;
>   };
> ```
> 
> been trying to avoid it, it's pressing my neck. 😛

Yes, it is a bit weird, do what you can and I will also have a look. The type system is really powerful but some times it can be a timesink to figure out how to make it work properly without force.

Reply:
Ok, will push in a bit.